### DISCUSSION TITLE
How to make a controlled DatePicker without a value

### DISCUSSION THREAD

**Discussion by gopeter** [adobe/react-spectrum#3664] [Q&A] [answered]
I'm trying to build a datepicker where I can choose an optional deadline, which means, that the date can be either a `DateValue` or undefined. But when it's undefined, React complains about moving from an uncontrolled to a controlled component. If I try to set the value to `null` (which works fine indeed!), `react-aria` complains about using null instead of `DateValue | undefined`.

Any ideas? 🤔

**Comment by ktabors** [comment, marked as answer]
Are you running in strict mode? There is an existing issue [#3187](https://github.com/adobe/react-spectrum/issues/3187).

**Reply by gopeter** [reply]
Oooh, didn't found that. Yes, I'm running strict mode

**Reply by ktabors** [reply]
We are working to support strict mode, but until then that will show up.
